pybossa/cache/projects.py

```diff
@@ -100,19 +100,27 @@ def format_task(row, lock_users=[]):
     args['user_id'] = user_id
     filters, filter_params = get_task_filters(args)
     filters += args.get("filter_by_wfilter_upref", {}).get("reserve_filter", "")
+    filters = filters.replace("task.", "mytable.").replace("coalesce(ct, 0)", "n_task_runs")
 
     sql = """
-            SELECT task.id,
-            coalesce(ct, 0) as n_task_runs, task.n_answers, ft,
-            priority_0, task.created, task.calibration,
-            task.user_pref, task.worker_filter, task.worker_pref
-            FROM task LEFT OUTER JOIN
-            (SELECT task_id, CAST(COUNT(id) AS FLOAT) AS ct,
-            MAX(finish_time) as ft FROM task_run
-            WHERE project_id=:project_id GROUP BY task_id) AS log_counts
-            ON task.id=log_counts.task_id
-            WHERE task.project_id=:project_id""" + filters + \
-            " ORDER BY %s" % (args.get('order_by') or 'id ASC')
+            SELECT id, n_task_runs, n_answers, ft, priority_0, created, calibration, user_pref, worker_filter, worker_pref
+            FROM ((
+                SELECT task.id, info, ct as n_task_runs, task.n_answers, ft, priority_0, task.created, task.state,
```

Comment:
Looks like we do not need select "info". Same for line 116.

Reply:
We need to select "info" because task browse page supports user filtering by task.info fields.

Reply:
How did the original query work without selecting "info"?

Reply:
The original query applies WHERE clause on the result of "task LEFT JOIN taskrun", which contains all columns from task table.
The new query applies WHERE clause on the result of (A UNION B), if A and B do not have info column, (A UNION B) won't contain info column.

Reply:
Makes sense.